programs/uxd/src/instructions/credix_lp/mint_with_credix_lp_depository.rs

```diff
@@ -0,0 +1,518 @@
+    );
+    msg!(
+        "[mint_with_credix_lp_depository:owned_shares_value_increase:{}]",
+        owned_shares_value_increase
+    );
+
+    // Validate that the locked value moved exactly to the correct place
+    require!(
+        user_collateral_amount_decrease == collateral_amount,
+        UxdError::CollateralDepositAmountsDoesntMatch,
+    );
+    require!(
+        user_redeemable_amount_increase == redeemable_amount_after_fees,
+        UxdError::CollateralDepositAmountsDoesntMatch,
+    );
+
+    // Check that we received the correct amount of shares
+    require!(
+        owned_shares_amount_increase == shares_amount,
+        UxdError::CollateralDepositAmountsDoesntMatch,
+    );
+    require!(
+        total_shares_amount_increase == shares_amount,
+        UxdError::CollateralDepositAmountsDoesntMatch,
+    );
+
+    // Check that the new state of the pool still makes sense
+    require!(
+        is_within_range_inclusive(
+            owned_shares_value_increase,
+            collateral_amount_after_precision_loss,
+            collateral_amount
+        ),
+        UxdError::CollateralDepositAmountsDoesntMatch,
+    );
+    require!(
+        is_within_range_inclusive(
+            total_shares_value_increase,
+            collateral_amount_after_precision_loss,
+            collateral_amount
+        ),
+        UxdError::CollateralDepositAmountsDoesntMatch,
+    );
+
+    // ---------------------------------------------------------------------
+    // -- Phase 4
+    // -- Emit resulting event, and update onchain accounting
+    // ---------------------------------------------------------------------
+
+    // Compute how much fees was paid
+    let redeemable_amount_delta: i64 =
+        compute_delta(redeemable_amount_before_fees, redeemable_amount_after_fees)?;
+    let minting_fee_paid: u64 = checked_i64_to_u64(-redeemable_amount_delta)?;
+
+    // Emit event
+    emit!(MintWithCredixLpDepositoryEvent {
+        controller_version: ctx.accounts.controller.load()?.version,
+        depository_version: ctx.accounts.depository.load()?.version,
+        controller: ctx.accounts.controller.key(),
+        depository: ctx.accounts.depository.key(),
+        user: ctx.accounts.user.key(),
+        collateral_amount: collateral_amount,
+        redeemable_amount: redeemable_amount_after_fees,
+        minting_fee_paid: minting_fee_paid,
+    });
+
+    // Accouting for depository
+    let mut depository = ctx.accounts.depository.load_mut()?;
+    depository.minting_fee_accrued(minting_fee_paid)?;
+    depository.collateral_deposited_and_redeemable_minted(
+        collateral_amount,
+        redeemable_amount_after_fees,
+    )?;
+
+    // Accouting for controller
+    let redeemable_amount_change: i128 = redeemable_amount_after_fees.into();
+    ctx.accounts
+        .controller
+        .load_mut()?
+        .update_onchain_accounting_following_mint_or_redeem(redeemable_amount_change)?;
+
+    // Done
+    Ok(())
```

Comment:
How much computing are the IXs btw?

Reply:
the mint is not so much, but the redeem is close to 180k out of 200k unit I believe, which removing the float operation makes it stay under the limit